src/common/hash/index.ts

```diff
@@ -30,6 +30,14 @@ const create = () => {
     delete deferredMap[msg.id];
     deferred?.resolve(msg);
   });
+  instance.on('exit', () => {
+    Object.keys(deferredMap).forEach(msgId => {
+      delete deferredMap[msgId];
+      const deferred = deferredMap[msgId];
+      deferred?.reject(new Error('hash.bundle.js process unexpectedly exited'));
+    });
```

Comment:
Alternately, we could actually issue those deferred requests again to the new process to avoid any kind of interruption

Reply:
Added some retrying with a max retry count